content/docs/tutorials/signing-and-verification.md

```diff
@@ -0,0 +1,757 @@
+
+- **RSASSA-PSS** (Probabilistic Signature Scheme) - Default and recommended
+- **RSASSA-PKCS1v15** - Legacy support
+
+### Directory Structure Setup
+
+First, create a consistent directory structure for your keys. This guide uses the following convention:
+
+```bash
+# Create a directory for your OCM keys
+mkdir -p ~/.ocm/keys
+
+# Recommended structure:
+# ~/.ocm/keys/
+#   ├── dev/          # Development keys
+#   ├── staging/      # Staging keys
+#   └── prod/         # Production keys
+```
+
+Throughout this guide, we'll use this structure and reference files with their full paths.
+
+### Self-Signed Keys (Development)
+
+Self-signed keys are ideal for development, testing, and environments without PKI infrastructure.
+
+#### Generate Development Keys
+
+```bash
+# Create directory for development keys
+mkdir -p ~/.ocm/keys/dev
+
+# Generate private key (4096 bits recommended)
+openssl genrsa -out ~/.ocm/keys/dev/private.key 4096
+
+# Extract public key
+openssl rsa -in ~/.ocm/keys/dev/private.key -pubout -out ~/.ocm/keys/dev/public.pem
+
+# Set secure permissions
+chmod 600 ~/.ocm/keys/dev/private.key
+chmod 644 ~/.ocm/keys/dev/public.pem
+```
+
+**Files created:**
+
+- `~/.ocm/keys/dev/private.key` - Private key (4096 bits)
+- `~/.ocm/keys/dev/public.pem` - Public key
+
+### CA-Signed Keys (Production)
+
+CA-signed certificates provide a trust chain that can be validated against a certificate authority.
+They are recommended for production environments, enterprise deployments,
+and scenarios requiring PKI integration or compliance.
+
+#### Generate Production Keys with CSR
+
+```bash
+# Create directory for production keys
+mkdir -p ~/.ocm/keys/prod
+
+# Generate private key
+openssl genrsa -out ~/.ocm/keys/prod/private.key 4096
+
+# Create certificate signing request (CSR)
+openssl req -new -key ~/.ocm/keys/prod/private.key \
+  -out ~/.ocm/keys/prod/request.csr \
+  -subj "/C=US/O=MyOrg/OU=Engineering/CN=OCM Production Signer"
+
+# Set secure permissions
+chmod 600 ~/.ocm/keys/prod/private.key
+chmod 644 ~/.ocm/keys/prod/request.csr
+
+# Send request.csr to your CA and save the response
+# After receiving from CA, save as:
+#   ~/.ocm/keys/prod/certificate.pem
+#   ~/.ocm/keys/prod/ca-chain.pem (if provided)
+```
+
+**Files created:**
+
+- `~/.ocm/keys/prod/private.key` - Private key (4096 bits)
+- `~/.ocm/keys/prod/request.csr` - Certificate signing request (send to CA)
+
+**Files received from CA:**
+
+- `~/.ocm/keys/prod/certificate.pem` - Signed certificate
+- `~/.ocm/keys/prod/ca-chain.pem` - CA certificate chain (optional but recommended)
+
+#### Create Certificate with Chain
+
+If you receive individual certificates, combine them into a chain:
+
+```bash
+# Combine leaf certificate with intermediate and root CA certificates
+cat ~/.ocm/keys/prod/certificate.pem \
+    ~/.ocm/keys/prod/intermediate-ca.pem \
+    ~/.ocm/keys/prod/root-ca.pem \
+    > ~/.ocm/keys/prod/cert-chain.pem
+
+chmod 644 ~/.ocm/keys/prod/cert-chain.pem
```

Comment:
This is a nice intro about signed key in general but I don't think it should be part of the tutorial on how OCM sign and verify works.